### ISSUE TITLE
Implement the sdk/metric/view package

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#2837]
Blocked by #2816 
Blocked by #2813 

- [ ] Add view match option functions
  - `func MatchInstrumentName(name string) Option`
  - `func MatchInstrumentNameRegexp(re *regexp.Regexp) Option`
  - `func MatchInstrumentKind(k sdkinstrument.Kind) Option`
  - `func MatchNumberKind(k number.Kind) Option`
  - `func MatchInstrumentationLibrary(lib instrumentation.Library) Option`
- [ ] Add instrument view option functions
  - `func WithKeys(keys ...attribute.Key) Option`
  - `func WithName(name string) Option`
  - `func WithDescription(desc string) Option`
  - `func WithAggregation(kind aggregation.Kind) Option`
  - `func WithTemporality(tempo aggregation.Temporality) Option`
  - `func WithAggregatorConfig(acfg aggregator.Config) Option`
- [ ] Add getter functions for the `Config` type
  - `func (Config) Name() string`
  - `func (Config) Keys() []attribute.Key`
  - `func (Config) Description() string`
  - `func (Config) Aggregation() aggregation.Kind`
  - `func (Config) Temporality() aggregation.Temporality`
  - `func (Config) AggregatorConfig() aggregator.Config`
- [ ] Add helper functions for the `Config type
  - `func (Config) IsSingleInstrument() bool`
  - `func (Config) HasName() bool`
  - `func (Config) Matches(lib instrumentation.Library, desc sdkinstrument.Descriptor) bool`

**Comment by MrAlias** [comment]
Only remaining task is to allow aggregation configuration. This has been split to its own issue: https://github.com/open-telemetry/opentelemetry-go/issues/2950
